subsys/bluetooth/host/l2cap.c

```diff
@@ -3035,6 +3039,94 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 	return 0;
 }
 
+#if defined(CONFIG_BT_L2CAP_RECONFIGURE_EXPLICIT)
+int bt_l2cap_ecred_chan_reconfigure_explicit(struct bt_l2cap_chan **chans, size_t chan_count,
+					     uint16_t mtu, uint16_t mps)
+{
+	struct bt_l2cap_ecred_reconf_req *req;
+	struct bt_conn *conn = NULL;
+	struct net_buf *buf;
+	uint8_t ident;
+
+	LOG_DBG("chans %p chan_count %u mtu 0x%04x mps 0x%04x", chans, chan_count, mtu, mps);
+
+	if (!chans || chan_count < 1 || chan_count > L2CAP_ECRED_CHAN_MAX_PER_REQ) {
```

Comment:
```suggestion
	if (!chans || !IN_RANGE(chan_count, 1, L2CAP_ECRED_CHAN_MAX_PER_REQ)) {
```

Reply:
We could also consider having a `L2CAP_ECRED_CHAN_MIN_PER_REQ = 1`

We could also consider moving this (these?) to the public API, as it is values that the application needs to know, and we can refer to these instead.

Reply:
I think 1 is obvious and does not need a define, since acting on an empty collection of channels does not make sense. We could return success instead of an error of course.

As for the maximum value, we can have a public define for that, but I think it should be renamed. This API does not say it results in exactly one request. Should it? If not then using a define with "_PER_REQ" in the define name with this API is inaccurate.

Reply:
> I think 1 is obvious and does not need a define, as it just means acting on an empty collection of channels does not make sense. 

That's fine, but IMO it makes more sense to do something like 
```
value range @ref L2CAP_ECRED_CHAN_MIN_PER_REQ to @ref L2CAP_ECRED_CHAN_MAX_PER_REQ
``` 

instead of 
```
value range 1 to @ref L2CAP_ECRED_CHAN_MAX_PER_REQ
```

At least to me it is kind of odd to mix and match defined values with literal values. 

> This API does not say it results in exactly one request. Should it? If not then using a define with "_PER_REQ" in the define name with this API is inaccurate.

Not explicitly, but the requirements that it imposes (max 5 channels, cannot decrease MPS if more than one channel, etc.) indicates a single L2CAP request. 

I won't block this PR based on this comment, so OK to keep as is. Was mainly a suggestion :)

Reply:
There are problems with a using a define like `L2CAP_ECRED_CHAN_MIN_PER_REQ = 1` in documentation.

 1. It's an extra level of indirection. If the programmer want's to know the actual value, they have to look it up.
 2. A symbol suggests it's an abstraction, and its value could change in the future. The programmer now has to worry about that possibility.

We could avoid both pitfall and still have a define by giving it a name that reflects it's true purpose. `MIN_SIZE_OF_NONEMPTY_COLLECTION = 1`. Is the following good?

    value range @ref MIN_SIZE_OF_NONEMPTY_COLLECTION to @ref L2CAP_ECRED_CHAN_MAX_PER_REQ

Reply:
You have a fair point. Looking at e.g. the ISO documentation, it does require an additional click to see the actual values of those defines: 

![image](https://github.com/user-attachments/assets/3078b7bc-6368-45f9-92ec-484d46df929e)

> A symbol suggests it's an abstraction, and its value could change in the future. The programmer now has to worry about that possibility.

That I am unsure that I agree on. It's very typical to use `#define`s for constant values that will never change. Our entire `hci_types.h` is built like that. 

> We could avoid both pitfall and still have a define by giving it a name that reflects it's true purpose. MIN_SIZE_OF_NONEMPTY_COLLECTION = 1. Is the following good?

I would prefer defined values for min and max to come from the same prefix if that makes sense. It's IMO a bit messy to do MIN from namespace A and MAX from namespace B, although I get your point. In this specific case I would actually prefer `1` over `MIN_SIZE_OF_NONEMPTY_COLLECTION`

Reply:
I see you already approved. To wrap up this thread: `MIN_SIZE_OF_NONEMPTY_COLLECTION` was a little absurd. I was actually expecting you to call me out on that. Instead I moved this requirement to the line above where it became the following:

```doxygen
@param chans Array of channels to reconfigure. Must be non-empty and
             contain at most 5 (@ref BT_L2CAP_ECRED_CHAN_MAX_PER_REQ)
             elements.
```